common/logger/logger.go

```diff
@@ -0,0 +1,32 @@
+// Copyright © 2020 NEC Laboratories Europe GmbH.
+//
+// Authors: Konstantin Munichev <konstantin.munichev@neclab.eu>
+//
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logger
+
+// Logger provides a basic functionality for any logger implementation
+type Logger interface {
+	Debug(args ...interface{})
+	Debugf(template string, args ...interface{})
+	Warning(args ...interface{})
+	Warningf(template string, args ...interface{})
+	Error(args ...interface{})
+	Errorf(template string, args ...interface{})
+	Info(args ...interface{})
+	Infof(template string, args ...interface{})
+	Fatal(args ...interface{})
+	Fatalf(template string, args ...interface{})
+}
```

Comment:
Why don't we order the methods by increasing severity level, i.e. debug, info, warning, error, fatal?